docs/capacity-planning.rst

```diff
@@ -0,0 +1,326 @@
+.. _ksql_capacity_planning:
+
+======================
+KSQL Capacity Planning
+======================
+
+.. contents::
+
+KSQL is a simple and powerful tool for building streaming applications on top of Kafka. This guide will help you get started with planning for provisioning your KSQL deployment by helping you answer questions like:
+
+    - What server specification should I use to run KSQL?
+    - Roughly how many KSQL server nodes will I need?
+    - Do I need to provision additional Kafka brokers to support KSQL?
+    - What kind of deployment mode makes sense for me?
+    - How do I know whether my KSQL queries are handling the incoming message rate, and how can I tune KSQL if it’s falling behind?
+
+Background
+==========
+
+Before jumping into the following sections, you may find it instructive to look through the `Kafka Streams capacity planning guide <https://docs.confluent.io/current/streams/sizing.html>`_. Under the hood KSQL uses the `Kafka Streams API <https://docs.confluent.io/current/streams/developer-guide/dsl-api.html>`_  to perform the actual query processing, so the details covered there apply to KSQL as well.
+
+Approach To Sizing
+==================
+
+This guide will help you get a rough idea of the computing resources required to run your KSQL queries. The emphasis here is that it’s a rough idea. As you’ll see in the following sections there are many factors at play and the best practice is always to stage and test your queries against realistic load first, and adjust your planned provisioning accordingly.
+
+General Guidelines
+==================
+
+KSQL
+----
+
+**CPU**: KSQL consumes CPU to serialize/deserialize messages into the declared stream/table schemas, and then process each message as required by the query. We would recommend at least 4 cores.
+
+**Disk**: KSQL uses local disk to persist temporary state for aggregations and joins. We recommend using an SSD and starting with at least 100GB.
+
+**Memory**: KSQL memory usage is dominated by on-heap message processing and off-heap state for aggregations (e.g., SUM, COUNT, TOPKDISTINCT) and joins. Usage for message processing scales with message throughput, while aggregation/join state is a function of topic partition count, key space size, and windowing. A good starting point here is 32GB.
+
+**Network**: KSQL relies heavily on Kafka, so fast and reliable network is important for optimal throughput. A 1Gib NIC is a good starting point.
+
+To recap, general guidelines for a basic KSQL server:
+    - 4 cores
+    - 32GB RAM
+    - 100GB SSD
+    - 1Gbit network
+
+Kafka
+-----
+
+KSQL consumes resources on your Kafka cluster.
+
+Internal Topics
++++++++++++++++
+
+KSQL creates the following types of topics on your KSQL cluster:
+
+**Output Topics**
+
+Every query started by a ``CREATE STREAM AS SELECT`` or ``CREATE TABLE AS SELECT`` statement writes its results to an output topic. The created topic is configured with the following properties:
+
```

Comment:
no need to indent bulleted items, otherwise an extra vertical bar is added:
![image](https://user-images.githubusercontent.com/11722533/39158349-65c5c9f2-4714-11e8-91d9-e1bb6263c560.png)